server/storage/sqldb.go

```diff
@@ -241,6 +289,7 @@ func (db *SQLStorage) Delete(gun data.GUN) error {
 		}
 		// if there weren't actually any records for the GUN, don't write
 		// a deletion change record.
+		// don't write a change if this isn't published yet
```

Comment:
Was this supposed to correspond to some channel manipulation?  (e.g. don't delete if the data isn't in the published channel?)